base/range.jl

```diff
@@ -58,6 +50,21 @@ function steprange_last{T}(start::T, step, stop)
     last
 end
 
+function steprange_last_empty{T<:Integer}(start::T, step, stop)
+    # empty range has a special representation where stop = start-1
+    # this is needed to avoid the wrap-around that can happen computing
+    # start - step, which leads to a range that looks very large instead
+    # of empty.
+    if step > zero(step)
+        last = start - one(stop-start)
+    else
+        last = start + one(stop-start)
+    end
+    last
+end
+# For types where x+one(x) may not be well-defined
+steprange_last_empty(start, step, stop) = start - step
```

Comment:
You are correct that you don't need the extra branch. May be worth adding a comment to the function.

> Or are you proposing this logic as a more general way to detect that the range is empty, without needing even `zero`?

I think the code using `zero` is cleaner and probably more performant.